packages/core/application/core-application-browser-internal/integration_tests/application_service.test.tsx

```diff
@@ -87,6 +90,38 @@ describe('ApplicationService', () => {
 
         expect(await currentAppId$.pipe(take(1)).toPromise()).toEqual('app1');
       });
+
+      it('updates the page_url analytics context', async () => {
+        const { register } = service.setup(setupDeps);
+
+        const context$ = setupDeps.analytics.registerContextProvider.mock.calls[0][0]
+          .context$ as Observable<{
+          page_url: string;
+        }>;
+        const locations: string[] = [];
+        context$.subscribe((context) => locations.push(context.page_url));
+
+        register(Symbol(), {
+          id: 'app1',
+          title: 'App1',
+          mount: async () => () => undefined,
+        });
+        register(Symbol(), {
+          id: 'app2',
+          title: 'App2',
+          mount: async () => () => undefined,
+        });
+
+        const { getComponent } = await service.start(startDeps);
+        update = createRenderer(getComponent());
+
+        await navigate('/app/app1/bar?hello=dolly');
+        await flushPromises();
+        await navigate('/app/app2#/foo');
+        await flushPromises();
+
```

Comment:
Q: would it work when navigating inside the same app?

Reply:
Yeah it does, but I can add a test to ensure that